docs/src/main/tut/ssl-tls.md

```diff
@@ -0,0 +1,167 @@
+---
+layout: docs
+title: SSL/TLS
+permalink: /docs/rpc/ssl-tls
+---
+
+# SSL/TLS Encryption
+
+> [gRPC](https://grpc.io/docs/guides/auth.html) has SSL/TLS integration and promotes the use of SSL/TLS to authenticate the server, and encrypt all the data exchanged between the client and the server. Optional mechanisms are available for clients to provide certificates for mutual authentication.
+
+[frees-rpc] allows you to encrypt the connection between the server and the client through SSL/TLS. The main goal of using SSL is to protect your sensitive information and keeps your data secure between servers and clients.
+
+As we mentioned in the [Quickstart](/docs/rpc/quickstart) section, we can choose and configure our client with `OkHttp` or `Netty` but if we want to encrypt our service, it's mandatory to use `Netty`.
+
+[frees-rpc] only supports encryptation over *Netty*.
+
+## Requirements 
+
+On the server and client side, we will need two files to configure the `SslContext` in `gRPC`:
+
+* Server/Client certificate file: Small data files that digitally bind a cryptographic key to an organization’s details. This file could be generated or obtained from a third company.
+
+* Server/Client private key file: The private key is a separate file that is used in the encryption of data sent between your server and the clients. All SSL certificates require a private key to work.
+
+## Usage
+
+The first step to do, in order to start to secure ours `freestyle-rpc` services, is use the artifacts `frees-rpc-netty-ssl` and `frees-rpc-client-netty` in our `build.sbt`.
+
+In second place, we will have to move to our `resources` folder both server/client certificates and private keys. 
+
+In our tests, the certificates have been extracted from [here](https://github.com/grpc/grpc-java/tree/master/testing/src/main/resources/certs). It could be a good point to start to check that our application works meanwhile we generate or obtain our own certificates.
+
+Thirdly, let's see a piece of code where we will explain line by line, what we are doing on the server side.
+
+We are going to ignore the explanations related to create the `RPCService`, `ServerRPCService` and runtime implicits. If you have any doubt about it, please, take a look at the [Patterns](/docs/rpc/patterns) section.
```

Comment:
We won't cover the details regarding creation of `RPCService`, `ServerRPCService` and runtime implicits. You can find more information about these in the [Patterns](/docs/rpc/patterns) section.